### DISCUSSION TITLE
Override getFilterExpression function

### DISCUSSION THREAD

**Discussion by thaingo** [yahoo/elide#2188] [Elide 5]
Lets take a look at the `value` variable  in the sample code below:
```
override fun getFilterExpression(entityClass: Type<*>, requestScope: RequestScope): FilterPredicate {
        val userPath = Path.PathElement(getClassType(User::class.java), STRING_TYPE, "id")
        val path = Path(listOf(userPath))
        **val value = listOf(requestScope.user.name)**
        return FilterPredicate(path, Operator.IN, value)
    }
```
Is it possible and advisable to fetch some value list from datastore or database and assign them to the `value` variable? 

It seems that only some predefined values (boolean, enum, hardcoded string) or/and extracted values from `RequestScope` can be used as `List<Object>` that is needed to build  `FilterPredicate`.

Appreciate your advice.

**Comment by aklish** [comment]
Yes, but put them in your subclass of User object.

You can have User return a list of something fetched from the DB.

**Reply by thaingo** [reply]
> Yes, but put them in your subclass of User object. You can have User return a list of something fetched from the DB.

Assume that we have `SubclassUser.class` (extends User) with its new field `geoLocation`, how to have User return a list of geoLocation values? I still don’t know how to write code to fetch the locations from DB.

Also want to clarify further:
- if User is a member of a api graph of many entities, does your above guidance still hold true for any member in that graph? Would I just have another subclass of another member entity and have it return another list of values fetched from the DB as well?
- How scale is this approach? What is your advice if size of the list of values is about 100+ KB?

**Comment by aklish** [comment]
To be clear, when I say 'User', I mean an Elide User object or principal which is different than a JPA bean or Elide model.   It is like a Spring Authentication object.

The filter expression check here is related to security - and the idea is that the User/Principal should contain (preloaded or lazily loaded) all the information about the user that any check might need or want to evaluate security rules.  

To populate our User/Principal, we just use straight JPA to fetch relevant info from the database.   This is typically done at the time of authentication (although it could be done later).

**Reply by thaingo** [reply]
Understood. Thanks @aklish .
